File: ctci/javascript/chapter1/Solution_1_1.js
var assert = require('assert');

function isUnique(word) {
  var chars = [];
for (var i = 0; i < word.length; i++) {
if (chars.indexOf(word[i]) > -1) {
return false;
}
chars.push(word[i]);
}
return true;
}
assert.equal(isUnique('apple'), false);
assert.equal(isUnique('true'), true);
assert.equal(isUnique('A'), true);
assert.equal(isUnique(''), true);

File: ctci/java/chapter1/Solution_1_1.java
package chapter1;

public class Solution_1_1 {
	public static boolean isUnique(String str) {
		if(str.length() > 256) {
			return false;
		}
		int checker = 0;
		for(int i = 0; i< str.length(); i++){
			int val = str.charAt(i);
			if((checker & (1 << val)) > 0){
				return false;
			}
			checker |= (1 << val);
		}
		return true;
	}

	public static void main(String[] args) {
		String[] words = {"apple", "peach","0123","01230"};
		for(String word : words){
			System.out.println(word + ":" + isUnique(word));
		}
	}
}
